src/Test.Utilities/DiagnosticAnalyzerTestBase.cs

```diff
@@ -430,21 +433,17 @@ protected static Project CreateProject(string[] sources, string language = Langu
 
             ProjectId projectId = ProjectId.CreateNewId(debugName: projectName);
 
+            var defaultReferences = ReferenceAssemblies.NetFramework.Net48.Default;
+            var references = Task.Run(() => defaultReferences.ResolveAsync(language, CancellationToken.None)).GetAwaiter().GetResult();
+
 #pragma warning disable CA2000 // Dispose objects before losing scope - Current solution/project takes the dispose ownership of the created AdhocWorkspace
             Project project = (addToSolution ?? new AdhocWorkspace().CurrentSolution)
 #pragma warning restore CA2000 // Dispose objects before losing scope
                 .AddProject(projectId, projectName, projectName, language)
-                .AddMetadataReference(projectId, MetadataReferences.CorlibReference)
-                .AddMetadataReference(projectId, MetadataReferences.SystemCoreReference)
-                .AddMetadataReference(projectId, AdditionalMetadataReferences.SystemXmlReference)
-                .AddMetadataReference(projectId, MetadataReferences.CodeAnalysisReference)
-                .AddMetadataReference(projectId, AdditionalMetadataReferences.SystemRuntimeFacadeRef)
-                .AddMetadataReference(projectId, AdditionalMetadataReferences.SystemThreadingFacadeRef)
-                .AddMetadataReference(projectId, AdditionalMetadataReferences.SystemThreadingTaskFacadeRef)
+                .AddMetadataReferences(projectId, references)
+                .AddMetadataReference(projectId, AdditionalMetadataReferences.CodeAnalysisReference)
```

Comment:
Why is `AdditionalMetadataReferences.CodeAnalysisReference` added by default here? We already have `ReferenceFlags.RemoveCodeAnalysis` check below and this Add should potentially move inside that if statement. It is also not clear to me what is `AdditionalMetadataReferences.CSharpSymbolsReference`? Can you clarify which assembly reference that is pulling in?

Reply:
I'll have a look but if I recall correctly I simply copy-pasted the content of the class from master so I am afraid I won't be able to explain much.

Reply:
@mavasani I confirm that I simply copy-pasted the full method from the class in master when the code has started to fail after the bump of version for the test framework.

Reply:
Sounds good. Might be good to create a separate tracking issue where you can just point to all the workarounds and unresolved comments in this PR so we can move forward with this PR, while not losing track of these issues.

Reply:
See #3089